apps-rendering/src/components/LiveBlocks/index.tsx

```diff
@@ -30,12 +32,27 @@ const contributorToBlockContributor = (
 interface LiveBlocksProps {
 	blocks: LiveBlock[];
 	format: ArticleFormat;
+	pinnedPost: Option<LiveBlock>;
+	pageNumber: number;
 }
 
-const LiveBlocks: FC<LiveBlocksProps> = ({ blocks, format }) => {
+const LiveBlocks: FC<LiveBlocksProps> = ({
+	blocks,
+	format,
+	pinnedPost,
+	pageNumber,
+}) => {
+	const pinnedPostId =
+		pinnedPost.kind === OptionKind.Some ? pinnedPost.value.id : '';
```

Comment:
there might be a neater way to express this (I'm still very new to the AR codebase!), but this might be a good opportunity to use [`withDefault`](https://github.com/guardian/dotcom-rendering/blob/56997678c39261f489651da147775cfabca2ec77/apps-rendering/src/option.ts#L52) and [`map`](https://github.com/guardian/dotcom-rendering/blob/56997678c39261f489651da147775cfabca2ec77/apps-rendering/src/option.ts#L73):

```
const pinnedPostId =
		compose(withDefault(''), map(pinnedPost => pinnedPost.id))(pinnedPost)
```

Reply:
Very interesting, I made the change 👍

Reply:
Oh nice spot George! This is an interesting discussion. I think for this kind of thing it's reasonable to use either approach, whichever you find easier to reason about. You'll see examples of `map`, `if` and `switch...case` used to work with these elsewhere in the AR codebase, and I think you'll also see `if (isSome)`, `match` and `map` used in Scala codebases.

@marjisound I'm guessing TS has probably pushed you to add a type annotation in your implementation of George's suggestion, because it struggles to infer the types with the argument order for `compose`? If you'd prefer to avoid this `pipe` may serve as an alternative:

```ts
pipe(
  pinnedPost,
  map(pinned => pinned.id),
  withDefault(''),
)
```

**Note:** Whatever option you go with I'd recommend using `pinned` (or some other name) over `pinnedPost` inside the function to avoid variable shadowing. Variable shadowing can sometimes cause problems: https://eslint.org/docs/latest/rules/no-shadow

Reply:
At the end because I took the approach of your other [suggestion](https://github.com/guardian/dotcom-rendering/pull/5372#discussion_r920339393) , I ended up not needing to derive the pinnedPostId any more. 

Your other suggestion:
```
isOriginalPinnedPost={pipe(
  pinnedPost,
  map((pinned) => block.id === pinned.id),
  withDefault<boolean>(false),
)}
```